node/pkg/lifecycle/startup/startup.go

```diff
@@ -573,6 +609,11 @@ func configureIPsAndSubnets(node *libapi.Node, k8sNode *v1.Node, getInterfaces f
 		return true, nil
 	}
 
+	if !reflect.DeepEqual(node.Spec.Interfaces, oldInterfaces) {
```

Comment:
How confident are you this list will remain sorted across multiple executions of calico/node? Might be worth explicitly sorting the returned interfaces so that we don't return `true` here arbitrarily.